runtime/vm/vmthread.c

```diff
@@ -1959,6 +1964,13 @@ startJavaThreadInternal(J9VMThread * currentThread, UDATA privateFlags, UDATA os
 	}
 	J9VMJAVALANGTHREAD_SET_THREADREF(currentThread, threadObject, newThread);
 
+#if (JAVA_SPEC_VERSION >= 14)
+    /* If thread was interrupted before start, make sure interrupt flag is set for running thread. */
+    if (J9VMJAVALANGTHREAD_DEADINTERRUPT(currentThread, threadObject)) {
+        omrthread_interrupt(osThread);
```

Comment:
Should this clear DEADINTERRUPT as well since it's being transferred to the native thread's interrupted state?  Otherwise, when the thread dies, it will appear to be interrupted as this will still be set

Reply:
I don't think its necessary since DEADINTERRUPT will be updated with the native state when the thread dies. I've made a comment over the deadInterrupt declaration in Thread.java to indicate that the state will not be updated when the thread is running.